hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1042,9 +1048,21 @@ func (r *HostedClusterReconciler) reconcileControlPlaneOperator(ctx context.Cont
 	}
 
 	// Reconcile operator deployment
+	var pullSecret corev1.Secret
+	if err := r.Client.Get(ctx, types.NamespacedName{Namespace: hcluster.Namespace, Name: hcluster.Spec.PullSecret.Name}, &pullSecret); err != nil {
+		return fmt.Errorf("failed to get pull secret: %w", err)
+	}
+	pullSecretBytes, ok := pullSecret.Data[corev1.DockerConfigJsonKey]
+	if !ok {
+		return fmt.Errorf("expected %s key in pull secret", corev1.DockerConfigJsonKey)
+	}
+	controlPlaneOperatorImage, err := getControlPlaneOperatorImage(ctx, hcluster, r.releaseProvider, r.HostedControlPlaneOperatorImage, pullSecretBytes)
+	if err != nil {
+		return err
+	}
 	controlPlaneOperatorDeployment := controlplaneoperator.OperatorDeployment(controlPlaneNamespace.Name)
 	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, controlPlaneOperatorDeployment, func() error {
-		return reconcileControlPlaneOperatorDeployment(controlPlaneOperatorDeployment, hcluster, r.HostedControlPlaneOperatorImage, controlPlaneOperatorServiceAccount)
```

Comment:
This breaks `hypershift-operator run --operator-image`, which I use to override the CPO, so I think that flag needs removed and moved over to `create cluster` which can then apply the new annotation

Reply:
Maybe `create cluster --control-plane-operator-image`

Reply:
If you use a 4.9 or 4.10 release image, it is not broken.  It would be broken for 4.8.  Let me make something up.

Reply:
The workflow is:

```
$ go run ./hypershift-operator run \
--operator-image quay.io/dmace/hypershift:noop \
--ignition-server-image quay.io/dmace/hypershift:latest

$ go run ./control-plane-operator run \
--in-cluster=false \
--namespace dmace-test \
--hosted-cluster-config-operator-image quay.io/dmace/hypershift:latest \
--etcd-operator-image quay.io/hypershift/etcd-cloud-operator:latest

$ hypershift create cluster dmace-test ...
```

Which allows running the hypershift operator and the CPO in local processes. The replacement with your PR would be:

```
$ go run ./hypershift-operator run \
--ignition-server-image quay.io/dmace/hypershift:latest

$ go run ./control-plane-operator run \
--in-cluster=false \
--namespace dmace-test \
--hosted-cluster-config-operator-image quay.io/dmace/hypershift:latest \
--etcd-operator-image quay.io/hypershift/etcd-cloud-operator:latest

$ hypershift create cluster dmace-test --control-plane-image quay.io/dmace/hypershift:noop ...
```

Reply:
Without the addition of a `--control-plane-image` flag in this PR, then after rebuilding `hypershift-operator`, I can no longer inject `quay.io/dmace/hypershift:noop` on the CLI because `create cluster` won't add the new annotation with the value of `--operator-image`.

Reply:
okay, thanks to @sjenning and @csrwng walking me through it again I see how this preserves the workflow. THANK YOU!